coap_packet_parse() returns different values.

sshahleg opened the pull request:
coap_packet_parse() returns different values depending on the errors.
1. EINVAL for input arg related errors.
2. EBADMSG for coap packet header related errors.
3. EILSEQ for errors related to parsing.

sagarshah-leg committed:
coap_packet_parse() updated to return different values for different errors

sagarshah-leg committed:
net: coap_packet_parse() updated to return different values other than EINVAL.

coap_packet_parse() now uses EINVAL for invalid input arguments, EBADMSG for
malformed coap packet header, and EILSEQ for malformed coap options.

Fixes: #48960

Signed-off-by: sagar.shah@legrand.us

sagarshah-leg committed:
updating comments for return values

sagarshah-leg committed:
net: updating comments for return values of coap_packet_parse()

Rewording API description for coap_packet_parse().

Fixes: #48960

Signed-off-by: sagar.shah@legrand.us

stephanosio reviewed (commented):
Please review the [Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines).

sagarshah-leg committed:
net: testing modified coap_packet_parse() API.

Adding test cases for testing return error values of coap_packet_parse()
under different scenarios. Updated existing test cases to reflect
expected return values.

Fixes: #48960

Signed-off-by: sagar.shah@legrand.us

sagarshah-leg committed:
net: Merge branch 'main' of github.com:sshahleg/zephyr-48960 into main

Merging to consolidate differences in commits.

Fixes: #48960

Signed-off-by: sagar.shah@legrand.us

sagarshah-leg committed:
Merge branch 'main' of github.com:sshahleg/zephyr-48960 into main

Merging to consolidate differences in commits.

Fixes: #48960

Signed-off-by: sagar.shah@legrand.us

sagarshah-leg committed:
net: Merge branch 'main' of github.com:sshahleg/zephyr-48960 into main

Merging branches to amend commit msg after code conflict.

Fixes: #48960

Signed-off-by: sagar.shah@legrand.us

sshahleg commented:
Closing and abandoning this PR due to code conflicts. Will create another one.
